spec/unit/transaction/resource_harness_spec.rb

```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env ruby
 require 'spec_helper'
 
-require 'puppet/transaction/resource_harness'
```

Comment:
I'm of the opinion that a class should require its immediate dependencies, even if that means a file is required multiple times. Ruby keeps track of which files have already been required (`LOADED_FEATURES`) so there's no harm in requiring multiple times. It will only be loaded once.

In cases where there is a cluster of related functionality, e.g. `lib/puppet/network`, we often just require the parent, with the knowledge that it requires all "related" files.

But in this case a `component` shouldn't know anything about `transaction`. If anything the dependency should be the other way around, as a transaction applies resources, which can include components.
 
The other reason is if you require `component` with the knowledge that it requires `transaction`, then you're making assumptions about transitive dependencies. If you ever make changes to `component` so it no longer requires `transaction`, then you'll break everything that depends on `component`. That seems very fragile.